payments-core/src/main/java/com/stripe/android/googlepaylauncher/GooglePayPaymentMethodLauncherViewModel.kt

```diff
@@ -0,0 +1,169 @@
+package com.stripe.android.googlepaylauncher
+
+import android.app.Application
+import androidx.annotation.VisibleForTesting
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.ViewModelProvider
+import androidx.lifecycle.distinctUntilChanged
+import com.google.android.gms.tasks.Task
+import com.google.android.gms.wallet.PaymentData
+import com.google.android.gms.wallet.PaymentDataRequest
+import com.google.android.gms.wallet.PaymentsClient
+import com.stripe.android.GooglePayConfig
+import com.stripe.android.GooglePayJsonFactory
+import com.stripe.android.Logger
+import com.stripe.android.PaymentConfiguration
+import com.stripe.android.model.PaymentMethodCreateParams
+import com.stripe.android.networking.ApiRequest
+import com.stripe.android.networking.StripeApiRepository
+import com.stripe.android.networking.StripeRepository
+import com.stripe.android.paymentsheet.DefaultGooglePayRepository
+import com.stripe.android.paymentsheet.GooglePayRepository
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.first
+import org.json.JSONObject
+import kotlin.coroutines.CoroutineContext
+
+internal class GooglePayPaymentMethodLauncherViewModel(
+    private val paymentsClient: PaymentsClient,
+    private val requestOptions: ApiRequest.Options,
```

Comment:
Not for this review, but I wonder if there are reasons not to include requestOptions in the StripeRepositoryConstructor?

Reply:
`Stripe` methods take a `stripeAccountId` and `idempotencyKey` so we can't know this ahead of time.

```
    @UiThread
    @JvmOverloads
    fun createPaymentMethod(
        paymentMethodCreateParams: PaymentMethodCreateParams,
        idempotencyKey: String? = null,
        stripeAccountId: String? = this.stripeAccountId,
        callback: ApiResultCallback<PaymentMethod>
    ) {
        executeAsync(callback) {
            stripeRepository.createPaymentMethod(
                paymentMethodCreateParams,
                ApiRequest.Options(
                    apiKey = publishableKey,
                    stripeAccount = stripeAccountId,
                    idempotencyKey = idempotencyKey
                )
            )
        }
    }
```

Reply:
It seems unlikely that they would change out the idempotencyKey, but maybe they do depending on the type of request?   Is the stripeAccountId changed when they need to use a different merchant based on the currency?

Reply:
I would expect `idempotencyKey` to be unique for requests that represent different things.

Reply:
what do you think about the stripeAccountId?

Reply:
A different `stripeAccountId` can be optionally passed in for every API request, see `createPaymentMethod()` above.

Reply:
Yes, this would be for users that have different accounts for different countries.